extensions/opentelemetry/opentelemetry/runtime/src/main/java/io/quarkus/opentelemetry/runtime/tracing/TracerRuntimeConfig.java

```diff
@@ -0,0 +1,40 @@
+package io.quarkus.opentelemetry.runtime.tracing;
+
+import java.util.List;
+import java.util.Optional;
+
+import io.quarkus.runtime.annotations.ConfigItem;
+import io.quarkus.runtime.annotations.ConfigPhase;
+import io.quarkus.runtime.annotations.ConfigRoot;
+
+@ConfigRoot(name = "opentelemetry.tracer", phase = ConfigPhase.RUN_TIME)
+public class TracerRuntimeConfig {
+
+    /**
+     * A comma separated list of name=value resource attributes that
+     * represents the entity producing telemetry
+     * (eg. {@code service.name=authservice}).
+     */
+    @ConfigItem
+    Optional<List<String>> resourceAttributes;
+
+    /**
+     * The sampler to use for tracing
+     * <p>
+     * Valid values are {@code always_off, always_on, parentbased_always_off,
```

Comment:
This merges two different aspects into a lengthy set of config options that might be confusing.

It would be clearer if the options were `on`, `off`, `ratio`, and we validate if `ratio` is set that `samplerRatio` is also set.

And then separately we have a `parentBased` which defaults to false.

We end up with the same 6 options, but spread over two separate properties reduces the duplication and possible confusion

Do you agree?

Reply:
Yup =]

Reply:
The only thing is that `parentBased` would be defaulted to `true` since the default is parent based always on =]

Reply:
Wow, didn't realize that was the default, I just knew it was `on`, not that it was `parentBased` as well.

That's fine

Reply:
Just updated it, if you can take another look =]